src/internal/connector/onedrive/item_test.go

```diff
@@ -257,3 +258,72 @@ func (suite *ItemIntegrationSuite) TestDriveGetFolder() {
 		})
 	}
 }
+
+func getPermsUperms(permID, userID string, scopes []string) (models.Permissionable, UserPermission) {
+	identity := models.NewIdentity()
+	identity.SetAdditionalData(map[string]any{"email": &userID})
+
+	sharepointIdentity := models.NewSharePointIdentitySet()
+	sharepointIdentity.SetUser(identity)
+
+	perm := models.NewPermission()
+	perm.SetId(&permID)
+	perm.SetRoles([]string{"read"})
+	perm.SetGrantedToV2(sharepointIdentity)
+
+	uperm := UserPermission{
+		ID:    permID,
+		Roles: []string{"read"},
+		Email: userID,
+	}
+
+	return perm, uperm
+}
+
+func TestOneDrivePermissionsFilter(t *testing.T) {
+	permID := "fakePermId"
+	userID := "fakeuser@provider.com"
+	userID2 := "fakeuser2@provider.com"
+
+	readPerm, readUperm := getPermsUperms(permID, userID, []string{"read"})
+	readWritePerm, readWriteUperm := getPermsUperms(permID, userID2, []string{"read", "write"})
+
+	noPerm, _ := getPermsUperms(permID, userID, []string{"read"})
+	noPerm.SetGrantedToV2(nil) // eg: link shares
+
+	cases := []struct {
+		name   string
+		perms  []models.Permissionable
+		uperms []UserPermission
```

Comment:
nit: it would be nice if this were `expectedPerms` or something else that denoted it's an expected result. Right now it's hard to tell if it's what the result should be or an input for the test

Reply:
`uperms` is essentially the `expectedPerms` unless I am missing something. Let me change the name.